staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -196,7 +200,9 @@ func (w *WebhookAuthorizer) Authorize(attr authorizer.Attributes) (decision auth
 		if r.Status.Allowed {
 			w.responseCache.Add(string(key), r.Status, w.authorizedTTL)
 		} else {
-			w.responseCache.Add(string(key), r.Status, w.unauthorizedTTL)
+			if len(key) <= maxUnauthorizedCachedKeySize {
```

Comment:
the only part that gives me pause is group membership, which could be large. ideally, we could separate out the parts under the caller's control (the ResourceAttributes/NonResourceAttributes)

Reply:
Done.

Reply:
This may be stupid observation, but the key itself can still be unusually large still, no? Do we want to exclude the user's groups prior to generating the cache key?

Reply:
> Do we want to exclude the user's groups prior to generating the cache key?

no, the cache key must reflect all inputs to the authorization call

Reply:
Then isn't `maxUnauthorizedCachedKeySize` a misnomer?